device/api/umd/device/coordinate_manager.h

```diff
@@ -36,7 +37,9 @@ class CoordinateManager {
         tt::ARCH arch,
         const size_t tensix_harvesting_mask = 0,
         const size_t dram_harvesting_mask = 0,
-        const size_t eth_harvesting_mask = 0);
+        const size_t eth_harvesting_mask = 0,
+        const BoardType board_type = BoardType::UNKNOWN,
```

Comment:
Consider changing to BlackholeCoreType.

Also consider splitting into two functions (one accepts this enum, one does not) if you don't want to have something like BlackholeCoreType::UNKNOWN.

Reply:
I think BlackholeCoreType is a bit an overkill in this situation, it may be useful when we have more complex topologies with Blackhole. I think even galaxy will be assembled from a lot of P150 board types, so I think this should be enough for now